src/components/study-pane.js

```diff
@@ -178,15 +181,22 @@ const StudyPane = () => {
 
     const sldRef = useRef();
     const handleUpdateSwitchState = useCallback( (breakerId, open, switchElement) => {
-        switchElement.querySelector(".open").style.visibility = open ? "visible" : "hidden";
-        switchElement.querySelector(".closed").style.visibility = open ? "hidden" : "visible";
+        let eltOpen = switchElement.querySelector(".open");
+        let eltClose = switchElement.querySelector(".closed");
+
+        eltOpen.style.visibility = open ? "visible" : "hidden";
+        eltClose.style.visibility = open ? "hidden" : "visible";
 
         updateSwitchState(studyName, breakerId, open).then( response => {
             if (response.ok) {
                 sldRef.current.reloadSvg();
+                sldRef.current.updateSwitchMsg("");
             }
             else {
                 console.error(response);
+                eltOpen.style.visibility = open ? "hidden" : "visible";
```

Comment:
Is this really necessary ? or shouldn't we just redownload the svg ?

Reply:
I preferred reverting the switch modification this way : it's easy and it doesn't add time overhead by reloading the entiere svg